pubsub/pubsub_test.go

```diff
@@ -0,0 +1,94 @@
+package pubsub_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/tendermint/tmlibs/pubsub"
+	"github.com/tendermint/tmlibs/pubsub/query"
+)
+
+func TestSubscribe(t *testing.T) {
+	t.Parallel()
+
+	s := pubsub.NewServer(1)
+	s.Start()
+	defer s.Stop()
+
+	ch := s.Subscribe(query.Empty{})
+	s.Publish("Goblin Queen")
+	assertReceive(t, "Goblin Queen", ch)
+}
+
+func TestQueries(t *testing.T) {
+	t.Parallel()
+
+	s := pubsub.NewServer(1)
+	s.Start()
+	defer s.Stop()
+
+	ch1 := s.Subscribe(q("tm.events.type=NewBlock"))
+	s.PublishWithTags("Iceman", map[string]interface{}{"tm.events.type": "NewBlock"})
+	assertReceive(t, "Iceman", ch1)
+
+	ch2 := s.Subscribe(q("tm.events.type=NewBlock AND abci.account.name=Igor"))
+	s.PublishWithTags("Iceman", map[string]interface{}{"tm.events.type": "NewBlock", "abci.account.name": "Igor"})
+	assertReceive(t, "Iceman", ch2)
+
+	ch3 := s.Subscribe(q("tm.events.type=NewBlock AND abci.account.name=Igor"))
+	s.PublishWithTags("Iceman", map[string]interface{}{"tm.events.type": "NewBlock"})
+	<-ch1
+	assert.Zero(t, len(ch3))
+}
+
+func TestPublish(t *testing.T) {
+	t.Parallel()
+
+	s := pubsub.NewServer(1)
+	s.Start()
+	defer s.Stop()
+
+	ch := s.Subscribe(query.Empty{})
+	s.Publish("Ka-Zar")
```

Comment:
Again, please two different names, so we can ensure that they are received in the same order they are sent.

Also, can you test what happens when eg. 50 events are sent before the client listens?  What is the capacity of the queue and what happens when it is full?